src/execution/join_hashtable.cpp

```diff
@@ -57,11 +60,55 @@ JoinHashTable::JoinHashTable(BufferManager &buffer_manager, const vector<JoinCon
 	idx_t block_capacity = MaxValue<idx_t>(STANDARD_VECTOR_SIZE, (Storage::BLOCK_SIZE / entry_size) + 1);
 	block_collection = make_unique<RowDataCollection>(buffer_manager, block_capacity, entry_size);
 	string_heap = make_unique<RowDataCollection>(buffer_manager, (idx_t)Storage::BLOCK_SIZE, 1, true);
+	swizzled_block_collection = block_collection->CloneEmpty();
```

Comment:
I kept getting burned by `keep_pinned` not transferring. Don't know if that is a problem here.

Reply:
I think the `swizzled_block_collection` does not need to have the `keep_pinned` flag no?

The idea of the `swizzled_block_collection` is that it is spillable, and does not need to keep the blocks pinned.